ksql-engine/src/test/java/io/confluent/ksql/engine/SqlFormatInjector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.engine;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.SqlFormatter;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.statement.Injector;
+import java.util.Objects;
+
+public class SqlFormatInjector implements Injector {
+
+  private final KsqlExecutionContext executionContext;
+
+  public SqlFormatInjector(final KsqlExecutionContext executionContext) {
+    this.executionContext = Objects.requireNonNull(executionContext, "executionContext");
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T extends Statement> ConfiguredStatement<T> inject(
+      final ConfiguredStatement<T> statement
+  ) {
+    try {
+      final Statement node = statement.getStatement();
+      final String sql = SqlFormatter.formatSql(node);
+      final String sqlWithSemiColon = sql.endsWith(";") ? sql : sql + ";";
```

Comment:
surely not ending with semi-colon is a bug in the formatter that should be fixed?

Reply:
I wasn't sure if this was intentional or not (e.g. some sql statements might be legitimate partial statements) but I think it might be okay to fix it at that layer. I will go ahead and do that and let you know if anything bad happens

Reply:
Actually after looking at it, it's a little bit more complicated than just that. Given that this targets 5.3 I'm going to leave this for another time.